### PR TITLE
#6142 Added abbility to see which plugins attached to the event #modxbughunt

### PR DESCRIPTION
**Pull request by alroniks**
### What does it do?
It allows to see which plugins attached to system event on the events page.

### Why is it needed?
When in the system a lot of plugins sometimes it can be useful so see which of them attached to events from events page. Or when we try to see which plugins work with specific event.

system settings - modx revolution

### Related issue(s)/PR(s)
Partially fix for https://github.com/modxcms/revolution/issues/6142

### CODE REVIEW

File: core/model/modx/processors/system/event/getlist.class.php [hunk 1]
```diff
@@ -11,60 +11,94 @@
...
         $list = array();
         /** @var modEvent $event */
         foreach ($data['results'] as $event) {
             $eventArray = $event->toArray();
 
+            if ($eventArray['plugins']) {
+                $c = $this->modx->newQuery('modPlugin');
+                $c->leftJoin('modPluginEvent', 'modPluginEvent', 'modPluginEvent.pluginid = modPlugin.id');
+                $c->sortby('modPluginEvent.priority', 'ASC');
+                $c->where(['modPluginEvent.event' => $eventArray['name']]);
```
Comment (rtripault) [inline core/model/modx/processors/system/event/getlist.class.php]: While i would love to use the short array syntax, this would break compatibility to PHP 5.3.x

**Review by rtripault — CHANGES_REQUESTED** [review]
Added a small note about short array syntax (which is PHP 5.4+).

Also, i was wondering if we should display only active plugin (`where(array('disabled' => false))` or style disabled plugins differently (rendered in the grid).

Otherwise :+1:

File: core/model/modx/processors/system/event/getlist.class.php (hunk 1 above)
Comment (Alroniks) [inline core/model/modx/processors/system/event/getlist.class.php]: Agree, It's a hard to follow when it has become a habit. :)

**Comment by Alroniks** [comment]
@rtripault I thought about to add marks for plugins, like `Ace (disabled, locked), pdoTools (locked)` and to avoid to change styles. My assumptions, that here we will have only readable info and disabled mostly applicable to form elements.

**Comment by Mark-H** [comment]
This looks useful, but can't be merged until the short array syntax is fixed ;)

**Comment by Alroniks** [comment]
@Mark-H @rtripault Fixed short arrays and add statuses for plugins (see screnshot below)
system settings - modx revolution 1

**Review by rtripault — APPROVED** [review]

File: core/model/modx/processors/system/event/getlist.class.php [hunk 2]
```diff
@@ -11,60 +11,103 @@
...
+                        : $plugin['name'];
+
+                    $plugins[] = array('name' => $pluginName);
+                }
+                $eventArray['plugins'] = $plugins;
+            } else {
+                $eventArray['plugins'] = '';
+            }
+
+            $eventArray['plugins'] = json_encode($eventArray['plugins']);
```
Comment (rtripault) [inline core/model/modx/processors/system/event/getlist.class.php]: What is the benefit of json encoding here and having to parse the json client side later ?

**Comment by rtripault** [comment]
All working as expected.
Just added a question about json encoding a value that would be json encoded later anyway

File: core/model/modx/processors/system/event/getlist.class.php (hunk 2 above)
Comment (Alroniks) [inline core/model/modx/processors/system/event/getlist.class.php]: You are right, it's obsolete. I've fixed and removed this encoding/decoding.

**Comment by rtripault** [comment]
Many thanks for your work and patience ;)

**Comment by gpsietzema** [comment]
#modxbughunt #1point to @rtripault
